File: archive/t/typescript/Palindromic-Number.ts
function isPalindrome(num: number): boolean {
    const numStr = num.toString();
    const reversedNumStr = numStr.split('').reverse().join('');
    return numStr === reversedNumStr;
}

if (process.argv.length !== 3) {
    console.log("Usage: please input a non-negative integer");
} else {
    const input = process.argv[2];
    const numberToCheck = parseFloat(input);

    if (!isNaN(numberToCheck) && Number.isInteger(numberToCheck) && numberToCheck >= 0) {
        if (isPalindrome(numberToCheck)) {
            console.log("true");
        } else {
            console.log("false");
        }
    } else {
        console.log("Usage: please input a non-negative integer");
    }
}

File: archive/d/dart/palindromic-number.dart
//Issue 4971
void main(List<String> args){
  const String error_message= "Usage: please input a non-negative integer";
  if (args.isEmpty){
    print(error_message);
    return;
  }

  try{
    bool is_palindrome = true;
    int num_needed = int.parse(args[0]);

    if (num_needed.isNegative){
      print(error_message);
      return;
    }
 
   String original_string = num_needed.toString();
   String reverse_string = original_string.split('').reversed.join('') ;
   is_palindrome = (original_string == reverse_string);

    print(is_palindrome);
  }
  catch(e){
    print(error_message);
  }
}
